doc/user_guide/model.rst

```diff
@@ -775,43 +791,80 @@ gaussian noise and proceed to fit as in the previous example.
     >>> line.coefficients.std
     (0.0081710549764721901, 1.4117294994070277)
 
-Because the noise is heterocedastic, the least squares optimizer estimation is
-biased. A more accurate result can be obtained by using weighted least squares
-instead that, although still biased for poissonian noise, is a good
-approximation in most cases.
+Because the noise is heteroscedastic, the least squares optimizer estimation is
+biased. A more accurate result can be obtained with weighted least squares,
+where the weights are proportional to the inverse of the noise variance.
+Although this is still biased for Poisson noise, it is a good approximation
+in most cases.
 
 .. code-block:: python
 
    >>> s.estimate_poissonian_noise_variance(
-   ...     expected_value=hs.signals.Signal1D(np.arange(300)))
-   >>> m.fit()
+   ...     expected_value=hs.signals.Signal1D(np.arange(300))
+   ... )
+   >>> m.fit(weights="inverse_variance")
    >>> line.coefficients.value
    (1.0004224896604759, -0.46982916592391377)
    >>> line.coefficients.std
    (0.0055752036447948173, 0.46950832982673557)
 
```

Comment:
I've added a `set_noise_variance()` method for convenience.

> It would be good to mention that the variance is only uses with `method=="ls"`

Is this true? Currently it says:
```
    weights = None

    if weights == "inverse_variance":
        weights = self._convert_variance_to_weights()

    args = (self.signal()[np.where(self.channel_switches)], weights)
```
and then `args` is passed to the optimizer, be it `leastsq`, `odr`, `mpfit` or the general `minimize()`

Reply:
In that case, I think that there is a bug in there: the weights should only be applied with method is LS to turn it into WLS. If we apply the weights when using Poisson-ML we get wrong results. Applying weights probably makes sense with the Huber-loss function though...

Reply:
I'll tidy it up!